tests/Cookies/NativeCookieTest.php

```diff
@@ -18,13 +18,83 @@
  * @link       http://cartalyst.com
  */
 
-namespace Cartalyst\Sentinel\Tests\Cookies;
+namespace Cartalyst\Sentinel\Cookies;
 
+use Mockery as m;
 use PHPUnit\Framework\TestCase;
-use Cartalyst\Sentinel\Cookies\NativeCookie;
+
+function setcookie($name, $value, $expires, $path, $domain, $secure, $httponly)
+{
+    return NativeCookieTest::$globalFunctions->setcookie(
+        $name,
+        $value,
+        $expires,
+        $path,
+        $domain,
+        $secure,
+        $httponly
+    );
+}
 
 class NativeCookieTest extends TestCase
 {
+    public static $globalFunctions;
+
+    protected function setUp(): void
+    {
+        self::$globalFunctions = m::mock();
+    }
+
+    /**
+     * @runInSeparateProcess
+     * @test
+     * */
+    public function it_can_set_different_options_for_cookie()
+    {
+        $options = [
+            'name'      => 'foo',
+            'domain'    => 'bar',
+            'path'      => 'foobar',
+            'secure'    => true,
+            'http_only' => true,
+        ];
+        $cookie = new NativeCookie($options);
```

Comment:
Not sure we need to have the options as a temporary variable, seems it can go directly on the class constructor.